.github/workflows/e2e-versions.yml

```diff
@@ -14,13 +14,14 @@ on:
   workflow_dispatch:
 jobs:
   setup-java-major-versions:
+    if: ${{ matrix.distribution != 'microsoft' || matrix.version != '8' }}
```

Comment:
Looks like this is causing the [workflow](https://github.com/actions/setup-java/actions/runs/1492164152) to fail:

```
The workflow is not valid. .github/workflows/e2e-versions.yml (Line: 17, Col: 9): Unrecognized named-value: 'matrix'. Located at position 1 within expression: matrix.distribution != 'microsoft' || matrix.version != '8'
```

Reply:
Seems like `matrix` must not be usable in `if`?  Let me double check

Reply:
Yep, this is why.  `if` at the job level is evaluated before the matrix is expanded, so it can't reference the `matrix` context.  `name` is evaluated when the job runs, so that's why you can use it there.

Instead, you can use `exclude` to knock a leg out of the matrix: https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#example-excluding-configurations-from-a-matrix

Reply:
done.